foundry/config/mod.rs

```diff
@@ -281,62 +334,61 @@ pub struct Config {
 
 impl Config {
     pub fn miner_options(&self) -> Result<MinerOptions, String> {
-        let (reseal_on_own_transaction, reseal_on_external_transaction) = match self.reseal_on_txs.as_deref() {
-            Some("all") => (true, true),
-            Some("own") => (true, false),
-            Some("ext") => (false, true),
-            Some("none") => (false, false),
-            Some(x) => {
+        let (reseal_on_own_transaction, reseal_on_external_transaction) = match self.reseal_on_txs.as_str() {
+            "all" => (true, true),
+            "own" => (true, false),
+            "ext" => (false, true),
+            "none" => (false, false),
+            x => {
                 return Err(format!(
                     "{} isn't a valid value for reseal-on-txs. Possible values are all, own, ext, none",
                     x
                 ))
             }
-            None => unreachable!(),
         };
 
         Ok(MinerOptions {
-            mem_pool_size: self.mem_pool_size.unwrap(),
-            mem_pool_memory_limit: match self.mem_pool_mem_limit.unwrap() {
+            mem_pool_size: self.mem_pool_size,
+            mem_pool_memory_limit: match self.mem_pool_mem_limit {
                 0 => None,
                 mem_size => Some(mem_size * 1024 * 1024),
             },
             reseal_on_own_transaction,
             reseal_on_external_transaction,
-            reseal_min_period: Duration::from_millis(self.reseal_min_period.unwrap()),
+            reseal_min_period: Duration::from_millis(self.reseal_min_period),
         })
     }
 
     pub fn rpc_http_config(&self) -> RpcHttpConfig {
         // FIXME: Add interface, cors and hosts options.
         RpcHttpConfig {
-            interface: self.jsonrpc_interface.clone().unwrap(),
-            port: self.jsonrpc_port.unwrap(),
+            interface: self.jsonrpc_interface.clone().expect("We set a default value in the Config struct"),
```

Comment:
This too? Should better be replaced with a comment and `unwrap()`?

Reply:
I changed it to use `unwrap()`.